File: features/like/ts/Like.ts
import { int64 } from "./lib/less";

/**
 * 赞
 * @type db
 */
export class Like {

    /**
     * ID
     */
    id: int64 = 0

    /**
     * 目标ID
     * @index ASC
     */
    tid: int64 = 0

    /**
     * 项ID
     * @index ASC
     */
    iid: int64 = 0

    /**
     * 用户ID
     * @index ASC
     */
    uid: int64 = 0

    /**
     * 其他数据
     * @length -1
     */
    options: any

    /**
     * 创建时间
     */
    ctime: int64 = 0

}

File: features/like/like/Like.go
package like

import (
	"github.com/hailongz/golang/db"
)

type Like struct {
	db.Object
	Tid	int64	`json:"tid" name:"tid" title:"目标ID" index:"ASC"`
	Iid	int64	`json:"iid" name:"iid" title:"项ID" index:"ASC"`
	Uid	int64	`json:"uid" name:"uid" title:"用户ID" index:"ASC"`
	Options	interface{}	`json:"options,omitempty" name:"options" title:"其他数据" length:"-1"`
	Ctime	int64	`json:"ctime" name:"ctime" title:"创建时间"`
}

func (O *Like) GetName() string {
	return "like"
}

func (O *Like) GetTitle() string {
	return "赞"
}

